### ISSUE TITLE
Dml From File

### ISSUE THREAD

**Issue by MggMuggins** [buggins/dlangui#447]
Sorry, I've been opening a lot of issues lately...

Is there any way to put my dml into a separate file and load that at runtime instead of writing it out in my D code? The reason I ask is that the documentation has this prototype for `parseML`:
```D
Widget parseML(T)( 
  string code, 
  string filename = "", 
  Widget context = null 
);
```

**Comment by buggins** [comment]
Currently, there is no such functionality.
It makes sense to implement it.
It could be possible to choose one of markup files from resources depending on DPI, screen size, etc. like it's done in Android UI API.
As well, it may be easier to edit such file comparing to embedded string constant in .d code.

**Comment by MggMuggins** [comment]
I agree. I also really like the idea of choosing different DML based on screen size.

How does that string constant even work? The syntax is this:
```D
q{String Contents}
```
But that doesn't make a whole lot of sense to me.

**Comment by MggMuggins** [comment]
I'll open a PR sometime in the next few days with dml file support added in a rudimentary sort of way. I'd like to do it in such a way that the tokenizer uses the file as a stream instead of simply copying the whole thing into RAM like my hacky solution (`file.readln('\x00')`)...

It's probably best not to allow parseML to take both a filename and a string at the same time. Seems like that introduces more complexity than is necessary, as well as leading to some ambiguous cases where it's not clear what the user wants. I may in that case somehow overload `parseML` and make some changes to the tokenizer to allow for a "mode".

Any thoughts on API? I would recommend deprecating the current declaration of parseML (with the signature I provided above) and doing two like this (rather than passing a string as the filename, since that's ambiguous and doesn't have a clear solution in my mind):
```D
T parseML(T = Widget)(string code, Widget context = null);
T parseML(T = Widget)(File file, Widget context = null);
```
Any ideas?
